packages/react/src/components/ContextualMenu/ContextualMenu.base.tsx

```diff
@@ -725,8 +725,12 @@ class ContextualMenuInternal extends React.Component<IContextualMenuInternalProp
         };
         ariaLabellledby = id;
       } else {
-        headerContextualMenuItem = sectionProps.title;
-        ariaLabellledby = this._id + sectionProps.title.text?.replace(/\s/g, '');
+        // Since title is an IContextualMenuItem and text property is optional
+        // there is the possibility if text is undefined that the id would appear
+        // in the DOM with 'undefined' appended to the end of the id, breaking the aria-labelledby scenario.
+        const id = sectionProps.title.text ? this._id + sectionProps.title.text.replace(/\s/g, '') : this._id;
```

Comment:
Looking at this further, I don't think the current solution will work properly, because `this._id` could end up being used for multiple things on the page (which isn't allowed). Instead I think it would be better to use the key in all cases.
```suggestion
        const id = this._id + sectionProps.title.key.replace(/\s/g, '');
```

Reply:
fixed